go/internal/cbor/types.go

```diff
@@ -13,3 +13,22 @@ const (
 	TypeTag         = 0xc0
 	TypeOther       = 0xe0
 )
+
+// getMajorType returns the first 3 bits of the first byte representing cbor's major type.
+// https://www.rfc-editor.org/rfc/rfc8949.html#name-major-types
+func getMajorType(b byte) Type {
+	switch (b & 0xff) >> 5 {
```

Comment:
Isn't that then untyped ikf i'd just return that? I would like to have the type for better readability later in code. But I changed b & 0b11100000 for the switch case!

Reply:
You can cast it to `Type`.
```go
    return Type(b & 0b11100000)
```
This is safe because TypeXXX covers all the 8 possible values of `b & 0b11100000`.